src/fsharp/TastOps.fs

```diff
@@ -1552,19 +1552,22 @@ let isClassTy g ty =
     | ILTypeMetadata (_,td) -> (td.tdKind = ILTypeDefKind.Class)
     | FSharpOrArrayOrByrefOrTupleOrExnTypeMetadata -> isFSharpClassTy g ty
 
-let isRefTy g ty = 
-    isUnionTy g ty || 
-    (isTupleTy g ty && not (isTupleStructTy g ty)) || 
-    isRecdTy g ty || 
-    isILReferenceTy g ty ||
-    isFunTy g ty || 
-    isReprHiddenTy g ty || 
-    isFSharpObjModelRefTy g ty || 
-    isUnitTy g ty
-
 let isStructTy g ty = 
     (isAppTy g ty && (tyconOfAppTy g ty).IsStructOrEnumTycon) || isTupleStructTy g ty
 
+let isRefTy g ty = 
```

Comment:
I'd somehow rather we didn't use ``not (isStructTy g ty)`` and just made the predicates distinct.  Just adding 

    let isStructOrEnumTyconTy g ty = isAppTy g ty && (tyconOfAppTy g ty).IsStructOrEnumTycon
    let isStructTy g ty = isStructOrEnumTyconTy g ty  || isTupleStructTy g ty
    let isRefTy g ty = not (isStructOrEnumTyconTy g ty) && ...

should do.  ``isTupleStructTy g ty`` is always false in the current implementation and indeed you can remove it if you like.